### PR TITLE
Improve Kytos daemon mode

### LINKED ISSUE #487: unable to (clean) stop kytos daemon from command line
if the daemon is started in the background wising simply ``$ kytosd``, there is no *clean* way to stop it...
kytosd has no start/stop option, and ``kytos controller start/stop`` (https://github.com/kytos/kytos-utils/issues/58) is not implemented yet.

### PR DESCRIPTION
**Pull request by macartur**
- Improve console and fix shutdown with kill signal
- Create a console class to kytos
- Add KytosDaemon class
- Change dependence python-daemon to daemons
- Add start,stop and status command using daemon
- Fix kytos/kytos#487

### CODE REVIEW

**Review by diraol — APPROVED** [review]

**Review by diraol — CHANGES_REQUESTED** [review]
Please, update the CHANGELOG !

**Review by diraol — DISMISSED** [review]

**Review by cemsbr — APPROVED** [review]

File: kytos/core/daemon.py [hunk 1]
```diff
@@ -1,104 +1,42 @@
...
+
+    def run(self):
+        """Run the controller and keep alive."""
+        try:
+            self.controller.start()
+        except SystemExit as exception:
+            self.controller.log.error(exception)
+            self.controller.log.info("Kytos daemon start aborted.")
+        while True:
+            pass
```
Comment (cemsbr) [inline kytos/core/daemon.py]: This busy waiting is using 100% CPU.

**Review by cemsbr — CHANGES_REQUESTED** [review]

**Review by cemsbr — CHANGES_REQUESTED** [review]
It's using 100% CPU.

File: kytos/core/daemon.py [hunk 2]
```diff
@@ -1,104 +1,45 @@
...
+            result = "Stopped."
+        return result
+
+    def shutdown(self, signum):
+        """Stop the controller and shutdown the process."""
+        self.controller.stop()
+        while True:
+            if self.controller.status() == 'Stopped':
+                super().shutdown(signum)
+            sleep(0.02)
```
Comment (cemsbr) [inline kytos/core/daemon.py]: Please, change to 0.5 and rebase.

**Review by cemsbr — CHANGES_REQUESTED** [review]

**Review by cemsbr — APPROVED** [review]

**Comment by hdiogenes** [comment]
Needs a description and/or linked issue. Improve the daemon in which ways?

**Comment by beraldoleal** [comment]
@macartur please provide a better description.

**Pull request closed without merging**